src/boat/private/Config.nim

```diff
@@ -1,10 +1,77 @@
 ##
 ## Config
 ## ======
-## base configuration for Manifest and CaptainsLog
+## provides a parser and standard interface
 
 ##[
 ## TLDR
-- a manifest eventually becomes the captains log
-  - from another perspective, the captains log is the realization of a manifest
+- come back later
+
+links
+-----
+- [parsecfg](https://nim-lang.org/docs/parsecfg.html)
+
 ]##
+
+import std/[
+  os,
+  parsecfg,
+  strutils,
+]
+
+import BoatErrors, BoatConstants
+
+type Config* = ref object of RootObj
+  use*: string ## \
+    ## path on disk pointing to a manifest / or dir with a manifest
+    ## URL pointing to a *.ini file
+  #--- private ---#
+  parsed: parsecfg.Config ## \
+    ## the parsed config after loading
+  saved: bool ## \
+    ## true if self.use has been saved to disk
+
+proc parseLocalManifest*(self: Config, path: string = self.use): bool =
+  ## parse self.use to self.parsed
+  ## prefer calling self.load or self.reload for validation
+  self.parsed = loadConfig path
+  result = true
+
+proc localManifestIsValid*(self: Config, path: string = self.use): bool =
+  ## throws if manifest not found, cant be read, or errors during parsing
+  let pathInfo = path.getFileInfo
+  result = case pathInfo.kind
+    of pcFile, pcLinkToFile:
+      if fpUserRead notin pathInfo.permissions: raise filePermissionError
+      elif not path.endsWith manifestName: raise manifestNameError
+      elif not self.parseLocalManifest path: raise configParseError
+      else: true
+    of pcDir, pcLinkToDir: self.localManifestIsValid self.use / manifestName
+
+proc save*(self: Config): bool =
```

Comment:
remember self.use maybe different than the actual path
- e.g. user provides a directory
- e.g. user provides a remote location

this should follow the existing pattern of proc(self: config, path = self.use) 
- if self.use != path we know captains log should point all requests for self.use to path